contracts/src/LightClient.sol

```diff
@@ -292,4 +327,54 @@ contract LightClient is Initializable, OwnableUpgradeable, UUPSUpgradeable {
             )
         );
     }
+
+    /// @notice Update the address of the permissioned prover
+    /// @dev The `permissionedProver` address can only be updated when `permissionedProverMode` ==
+    /// true
+    /// If the new prover address is the same as the current one, the method reverts with the
+    /// `NoChangeRequired` error
+    function updatePermissionedProver(address prover) public onlyOwner {
+        // permissionedProverMode enabled and only update the permissionedProver if it's not the
+        // same as the current one
+        if (permissionedProverMode && prover != permissionedProver) {
+            if (prover == address(0)) {
+                revert InvalidAddress();
+            }
+            permissionedProver = prover;
+            emit PermissionedProverUpdated(prover);
+        } else {
+            revert NoChangeRequired();
+        }
+    }
+
+    /// @notice set the permissionedProverMode to true and set the permissionedProver to the
+    /// non-zero address provided
+    /// @dev if it was already enabled (permissioneProverMode == true), then revert with
+    /// NoChangeRequired
+    function enablePermissionedProverMode(address prover) public onlyOwner {
+        if (!permissionedProverMode) {
+            if (prover == address(0) && permissionedProver == address(0)) {
```

Comment:
Why is the condition `prover == address(0)` not enough?

Reply:
The two functions are now almost the same it's just that the `enable` function reverts if the permissioned prover mode is already enabled. I wonder if this is an important feature. My feeling is that it isn't and we can provide a single function to set the prover and enable permissioned prover mode (if it was previously disabled). cc @philippecamacho @alysiahuggins @alxiong

Reply:
@philippecamacho , you're right, the first condition is enough. 

As far as reverting if the permissionedProver is already set - there's another function `updatePermissionedProver` which allows the admin to update the permissioned prover. So are you suggesting that we get rid of the `updatePermissionedProver` function? If so, I agree, this isn't a function that should be called that many times anyway, so having multiple functions to set the prover with different logging doesn't seem necessary.

Reply:
consolidated the update and enable prover functions into one `setPermissionedProver` https://github.com/EspressoSystems/espresso-sequencer/commit/f521dac663d5563820273529913ebe88b86fa915